### ISSUE TITLE
Windows support

### ISSUE THREAD

**Issue by pavdmyt** [pavdmyt/yaspin#33]
In continuation to discussion on windows support for `yaspin` started in #32 

Minimum plan towards Windows support:

- apply the [patch](https://github.com/pypa/pipenv/blob/master/tasks/vendoring/patches/vendor/yaspin-signal-handling.patch) for handling signals
- use [vistir/cursor](https://github.com/sarugaku/vistir/blob/master/src/vistir/cursor.py) module for cursor support
- implement (or find the replacement for) [termcolor](https://pypi.org/project/termcolor/) features in Windows environment
- unittests for all `yaspin` features under Windows
- CI pipeline for tests under Windows

**Comment by techalchemy** [comment]
- It is definitely possible to retain the current API using colorama, I preserved the API for use in `vistir` with a wrapper and some tweaks to the implementation, so it shouldn't be a problem at all

**Comment by pavdmyt** [comment]
Related: #24

**Comment by oz123** [comment]
`pipenv` is currently getting new wind. I am currently helping to maintain pipenv, and I would be happy if we could upstream pipenv`s patches to yaspin. Would we be able to do so? Is there any way we can help you with the adoption of these patches?
@pavdmyt @matteius

**Comment by pavdmyt** [comment]
Hi @oz123 

What are these patches about? Generally, I'm open for the new PRs.

**Comment by pavdmyt** [comment]
@oz123 yes, I'm open to improve Windows support. Feel free to submit PRs.
